pkg/controller/workspace/workspace_controller.go

```diff
@@ -285,3 +290,12 @@ func getWorkspaceId(instance *workspacev1alpha1.Workspace) (string, error) {
 	}
 	return "workspace" + strings.Join(strings.Split(uid.String(), "-")[0:3], ""), nil
 }
+
+func hasTheiaEditor(components []workspacev1alpha1.ComponentSpec) bool {
```

Comment:
I would suggest `isCheRestApisRequired` as the name here; I've tried to name this type of utility function based on what it's used for rather than what it does (e.g. `isArtifactsBrokerNecessary` instead of `containsVsCodeExtension`). Not a big deal though.